src/transforms/sample/transform.rs

```diff
@@ -12,15 +12,44 @@ use crate::{
 use vector_lib::lookup::lookup_v2::OptionalValuePath;
 use vector_lib::lookup::OwnedTargetPath;
 
+/// Exists only for backwards compatability purposes so that the value of sample_rate_key is
+/// consistent after the internal implementation of the Sample class was modified to work in terms
+/// of percentages
+#[derive(Clone, Copy, Debug)]
+pub enum SampleRate {
+    OneOverN(u64),
+    Percentage(f32),
+}
+
+impl SampleRate {
+    pub fn pct(&self) -> f32 {
```

Comment:
nit: Abbreviations don't save money. 😛 Also, this can take `self` as owner since `Self : Copy`.